### PR TITLE
Remove implicits from tasty reflection

### PR DESCRIPTION
**Pull request by nicolasstucki**
Based on #8138

### CODE REVIEW

File: library/src/scala/tasty/Reflection.scala [hunk 1]
```diff
@@ -419,40 +419,40 @@ class Reflection(private[scala] val internal: CompilerInterface) { self =>
   //   QUOTES   //
   ////////////////
 
-  implicit class QuotedExprAPI[T](expr: scala.quoted.Expr[T]) {
+  given  QuotedExprOps: extension (expr: scala.quoted.Expr[?]) {
```
Comment (liufengyun) [inline library/src/scala/tasty/Reflection.scala]: Can we use the new syntax in #7917? Same for the changes blow.

File: library/src/scala/tasty/Reflection.scala [hunk 2]
```diff
@@ -419,40 +419,40 @@ class Reflection(private[scala] val internal: CompilerInterface) { self =>
...
   }
 
-  implicit class TypeToQuotedAPI(tpe: Type) {
-    /** Convert `Type` to an `quoted.Type[_]` */
-    def seal(given ctx: Context): scala.quoted.Type[_] =
-      internal.QuotedType_seal(tpe)
+  given TypeToQuotedOps: extension (tpe: Type) {
+      /** Convert `Type` to an `quoted.Type[_]` */
+      def seal(given ctx: Context): scala.quoted.Type[_] =
+        internal.QuotedType_seal(tpe)
```
Comment (liufengyun) [inline library/src/scala/tasty/Reflection.scala]: Minor: inconsistent indentation.

File: library/src/scala/tasty/Reflection.scala [hunk 3]
```diff
@@ -2742,7 +2742,7 @@ class Reflection(private[scala] val internal: CompilerInterface) { self =>
     def foldTrees(x: X, trees: Iterable[Tree])(given ctx: Context): X = trees.foldLeft(x)(foldTree)
 
     def foldOverTree(x: X, tree: Tree)(given ctx: Context): X = {
-      def localCtx(definition: Definition): Context = definition.symbol.localContext
+      lazy val localCtx: Context = tree.symbol.localContext
```
Comment (liufengyun) [inline library/src/scala/tasty/Reflection.scala]: Minor: I feel the previous version is more readable.

File: library/src/scala/tasty/Reflection.scala (hunk 1 above)
Comment (nicolasstucki) [reply library/src/scala/tasty/Reflection.scala]: I'll update that syntax in a following PR

Comment (nicolasstucki) [reply library/src/scala/tasty/Reflection.scala]: #8156

**Pull request merged**